deploy/charts/firefly/templates/core/statefulset.yaml

```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "firefly.fullname" . }}
+  labels:
+    {{- include "firefly.coreLabels" . | nindent 4 }}
+spec:
+  serviceName: {{ include "firefly.fullname" . }}
+  updateStrategy:
+    type: RollingUpdate
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "firefly.coreSelectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/core/secret.yaml") . | sha256sum }}
+        {{- with .Values.core.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "firefly.coreSelectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.core.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.core.podSecurityContext | nindent 8 }}
+      containers:
+        - name: firefly
+          securityContext:
+            {{- toYaml .Values.core.securityContext | nindent 12 }}
+          image: "{{ .Values.core.image.repository }}:{{ .Values.core.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.core.image.pullPolicy }}
+          env:
+            - name: FIREFLY_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          {{- if .Values.core.extraEnv }}
+            {{- toYaml .Values.core.extraEnv | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.core.service.httpPort }}
+              protocol: TCP
+            {{- if .Values.config.debugEnabled }}
+            - name: debug
+              containerPort: {{ .Values.core.service.debugPort }}
+              protocol: TCP
+            {{- end }}
+            {{- if .Values.config.adminEnabled }}
+            - name: admin
+              containerPort: {{ .Values.core.service.adminPort }}
+              protocol: TCP
+            {{- end }}
+          livenessProbe:
+            tcpSocket:
+              port: {{ if and .Values.config.adminEnabled .Values.config.preInit }}admin{{ else }}http{{ end }}
```

Comment:
Why is the admin port sometimes used for the liveness probe instead of always HTTP?

Reply:
If FF is in a `preInit` state the HTTP port is not accessible.

Although folks likely will never want to deploy it in a `preInit` state in a cloud-native setting, we gave them the option and so had to make the probes align with that.